analyzer/emerald/incoming.go

```diff
@@ -0,0 +1,365 @@
+		}
+		blockTransactionData.RelatedAccountAddresses = map[string]bool{}
+		tx, err := common.VerifyUtx(sigContext, &txr.Tx)
+		if err != nil {
+			err = fmt.Errorf("tx %d: %w", i, err)
+			fmt.Println(err)
+			tx = nil
+		}
+		if tx != nil {
+			blockTransactionData.SignerData = make([]*BlockTransactionSignerData, 0, len(tx.AuthInfo.SignerInfo))
+			for j, si := range tx.AuthInfo.SignerInfo {
+				var blockTransactionSignerData BlockTransactionSignerData
+				blockTransactionSignerData.Index = j
+				addr, err1 := registerRelatedAddressSpec(blockData.AddressPreimages, blockTransactionData.RelatedAccountAddresses, &si.AddressSpec)
+				if err1 != nil {
+					return nil, fmt.Errorf("tx %d signer %d visit address spec: %w", i, j, err1)
+				}
+				blockTransactionSignerData.Address = addr
+				blockTransactionSignerData.Nonce = int(si.Nonce)
+				blockTransactionData.SignerData = append(blockTransactionData.SignerData, &blockTransactionSignerData)
+			}
+			if err = common.VisitCall(&tx.Call, &txr.Result, &common.CallHandler{
+				AccountsTransfer: func(body *accounts.Transfer) error {
+					if _, err = registerRelatedSdkAddress(blockTransactionData.RelatedAccountAddresses, &body.To); err != nil {
+						return fmt.Errorf("to: %w", err)
+					}
+					return nil
+				},
+				ConsensusAccountsDeposit: func(body *consensusaccounts.Deposit) error {
+					if _, err = registerRelatedSdkAddress(blockTransactionData.RelatedAccountAddresses, body.To); err != nil {
+						return fmt.Errorf("to: %w", err)
+					}
+					return nil
+				},
+				ConsensusAccountsWithdraw: func(body *consensusaccounts.Withdraw) error {
+					// .To is from another chain, so exclude?
+					return nil
+				},
+				EvmCreate: func(body *evm.Create, ok *[]byte) error {
+					if !txr.Result.IsUnknown() && txr.Result.IsSuccess() && len(*ok) == 32 {
+						// todo: is this rigorous enough?
+						if _, err = registerRelatedEthAddress(blockData.AddressPreimages, blockTransactionData.RelatedAccountAddresses, common.SliceEthAddress(*ok)); err != nil {
+							return fmt.Errorf("created contract: %w", err)
+						}
+					}
+					return nil
+				},
+				EvmCall: func(body *evm.Call, ok *[]byte) error {
+					if _, err = registerRelatedEthAddress(blockData.AddressPreimages, blockTransactionData.RelatedAccountAddresses, body.Address); err != nil {
+						return fmt.Errorf("address: %w", err)
+					}
+					// todo: maybe parse known token methods
+					return nil
+				},
+			}); err != nil {
+				return nil, err
+			}
+		}
+		var txGasUsed int64
+		foundGasUsedEvent := false
+		if err = common.VisitSdkEvents(txr.Events, &common.SdkEventHandler{
+			Core: func(event *core.Event) error {
+				if event.GasUsed != nil {
+					if foundGasUsedEvent {
+						return fmt.Errorf("multiple gas used events")
+					}
+					foundGasUsedEvent = true
+					txGasUsed = int64(event.GasUsed.Amount)
+				}
+				return nil
+			},
+			Accounts: func(event *accounts.Event) error {
+				if event.Transfer != nil {
```

Comment:
Here and all handlers above: We don't need to do anything here (other than not the related address) because the event/tx is already handled by the consensus-level analyzer. Is that correct?

Reply:
didn't have any other functionality planned, no

Reply:
addendum: to my knowledge the consensus analyzer does not do anything with these events

Reply:
cool, I think we can do similar here with paratime accounts